sound/soc/sof/sof-audio.c

```diff
@@ -376,6 +376,9 @@ sof_prepare_widgets_in_path(struct snd_sof_dev *sdev, struct snd_soc_dapm_widget
 			p->walking = false;
 			if (ret < 0) {
 				/* unprepare the source widget */
+				if (!widget_ops)
+					return ret;
+
```

Comment:
I think the fix is rather this: https://github.com/thesofproject/linux/pull/4061